### PR TITLE
Update shuffle and swap

### PR DESCRIPTION
**Pull request by parantak**
shuffle and swap have been merged

### CODE REVIEW

**Comment by parantak** [comment]
@someshsingh22 and @rajaswa , could you help me with this? I added the unit tests and the formatting's been done. Am I missing something?

**Comment by someshsingh22** [comment]
I dont exactly know but there must be some problem in the coverage of unit cases
[Coverage Error Source](https://codecov.io/gh/SforAiDl/decepticonlp/compare/683cdba1f8dcdcab8f3f2ecbfa7de50fc970cce0...eda8049699ca6523b1ccce25d926b9b4ff39131c/src/decepticonlp/transforms/perturb.py?before=decepticonlp/transforms/perturb.py) Its failing here maybe @Sharad24 can help more

**Comment by parantak** [comment]
@someshsingh22 , could it possibly be that we'd have to add a test case for when ignore=True? Though, that did not pose an issue previously

File: tests/test_perturb.py [hunk 1]
```diff
@@ -44,22 +44,29 @@ def test_perturb_insert_space_with_character_size_less_than_two():
...
+    random.seed(0)
+    assert perturb.shuffle(word, mid=False) == expected_result
+
+
+@pytest.mark.parametrize(
+    "word, expected_result", [("Adversarial", "Aiavrsedarl"),],
+)
+def test_perturb_shuffle(word, expected_result):
+    random.seed(0)
+    assert perturb.shuffle(word) == expected_result
```
Comment (Sharad24) [inline tests/test_perturb.py:59]: Both tests have same names. So the final one overwrites the results of the first one.

File: decepticonlp/transforms/perturb.py [hunk 2]
```diff
@@ -58,14 +63,24 @@ def swap(word, ignore=True):
...
-    charlist = list(word)
-    index = random.randint(1, len(word) - 3)  # select random offset for tuple
-    charlist[index], charlist[index + 1] = (
-        charlist[index + 1],
-        charlist[index],
-    )  # swap tuple
-    return "".join(charlist)
+    ), "Word needs to have a minimum length of 4 for a shuffle operation"
+
+    if mid:
```
Comment (Sharad24) [inline decepticonlp/transforms/perturb.py:68]: `nit`
Make this `middle` instead of `mid`?

File: decepticonlp/transforms/perturb.py [hunk 3]
```diff
@@ -31,25 +31,30 @@ def insert_space(word, ignore=True):
...
-    :ignore: default (True), boolean if assertions should be ignored
-
-    -returns word with random character swaps
+    :param word : word to be shuffled
+    :param mid : 
+    if set, it shuffle all the characters barring the initial and last
+    if not set, it swap any two characters barring the initial and last 
+                
+    
+    returns shuffled word with first and last character intact
```
Comment (Sharad24) [inline decepticonlp/transforms/perturb.py:56]: `nit` 
Two statements on `returns` and `rtype` missing here. This is just for avoiding documenting work later on :)

Comment (Sharad24) [reply decepticonlp/transforms/perturb.py:56]: More info here: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

**Comment by Sharad24** [comment]
Also, you can run `pytest --cov='./decepticonlp' --cov-report=html` followed by `open htmlcov/index.html` to see code coverage locally.

File: tests/test_perturb.py (hunk 1 above)
Comment (parantak) [reply tests/test_perturb.py]: Resolved on a branch, commit pending.

**Comment by parantak** [comment]
@Sharad24 , could you see to the last test and help out. I fixed the overlapping test cases and I will fix the documentation soon. Thanks.

File: tests/test_perturb.py [hunk 4]
```diff
@@ -44,22 +44,29 @@ def test_perturb_insert_space_with_character_size_less_than_two():
...
-def test_perturb_swap(word, expected_result):
-    random.seed(42)
-    assert perturb.swap(word) == expected_result
+def test_perturb_shuffle_swap_two(word, expected_result):
+    random.seed(0)
+    assert perturb.shuffle(word, mid=False) == expected_result
+
+
+@pytest.mark.parametrize(
+    "word, expected_result", [("Adversarial", "Aiavrsedarl"),],
```
Comment (rajaswa) [inline tests/test_perturb.py:55]: We are missing the test case when ignore=True and we have a " " or length <4
Add this example in parametrize here: ("dog", "dog")

File: decepticonlp/transforms/perturb.py [hunk 5]
```diff
@@ -31,25 +31,30 @@ def insert_space(word, ignore=True):
...
+    
     """
     if ignore and (" " in word or len(word) < 4):
         return word
```
Comment (Sharad24) [inline decepticonlp/transforms/perturb.py:60]: It seems it's coming from never reaching this statement `return word`

File: tests/test_perturb.py (hunk 4 above)
Comment (rajaswa) [reply tests/test_perturb.py:55]: @parantak  I think this is the only change we need

**Comment by parantak** [comment]
All tests are done. Thanks.

**Comment by Sharad24** [comment]
Great!

**Pull request merged**